common/build/eslint-plugin-fluid/src/test/mockFiles/no-unchecked-record-access/fileWithOnlyRecordAccess.ts

```diff
@@ -0,0 +1,78 @@
+indexedRecordOfStrings[b].length; // defect: Accessing length of index property 'b' using bracket notation, but 'b' might not be present
+indexedRecordOfStrings.a?.length; // ok: Using optional chaining to access length safely handles 'undefined'
+indexedRecordOfStrings.a!.length; // ok: The author says they understand the question raised by check and acknowledge that they have other information expecting that it is actually defined or that they are okay with an exception being raise here if "a" is not present and defined
+indexedRecordOfStrings["a"]?.length; // ok: Using optional chaining to access length using bracket notation safely handles 'undefined'
+indexedRecordOfStrings["a"]!.length; // ok: The author says they understand the question raised by check and acknowledge that they have other information expecting that it is actually defined or that they are okay with an exception being raise here if "a" is not present and defined
+
+if (indexedRecordOfStrings.a) {
+	indexedRecordOfStrings.a.length; // ok: Within a truthy check, 'a' is guaranteed to be defined
+}
+
+if ("a" in indexedRecordOfStrings) {
+	indexedRecordOfStrings.a.length; // ok: Accessing length of property inside an 'in' check, 'a' is guaranteed to be defined
+}
+
+for (const [key, value] of Object.entries(indexedRecordOfStrings)) {
+	value.length; // ok: Object.entries provides only present values
+	indexedRecordOfStrings[key]; // ok: Accessing property while looping though records which acts like a `has` property check
+	indexedRecordOfStrings[key].length; // ok: Accessing property while looping though records which acts like a `has` property check
+}
+
+const aExpectingString: string = indexedRecordOfStrings.a; // defect: Assigning index property 'a' to a strict string variable, but 'a' might not be present
+const aExpectingStringOrUndefined: string | undefined = indexedRecordOfStrings.a; // ok: Assigning index property 'a' to string or undefined variable, 'a' might not be present
+const aImplicitType = indexedRecordOfStrings.a; // defect: Assigning index property with inferred type
+aImplicitType.length; // ok: This should be caught by tsc, not by this custom lint rule
+aImplicitType?.length; // ok: This should be caught by tsc, not by this custom lint rule
+aImplicitType!.length; // ok: This should be caught by tsc, not by this custom lint rule
+
+interface NonNullableProps {
+	definitelyString: string;
+	maybeString?: string;
+}
+
+const nonNullObj: NonNullableProps = { definitelyString: "hello" };
+nonNullObj.definitelyString.length; // ok: Accessing length of non-nullable property
+nonNullObj.maybeString.length; // ok: This should be caught by tsc, not by this custom lint rule
+
+let possiblyUndefined: string | undefined;
+possiblyUndefined = nonNullObj.maybeString; // ok: Assigning optional property to variable of type 'string | undefined'
```

Comment:
`let` is a good case to test as well.
Add test cases with let similar to lines 61-62. (I think you can skip a line 63 let type case as there should be a type associated with the variable from its initial assignment.)